sympy/solvers/solveset.py

```diff
@@ -1117,3 +1120,514 @@ def linsolve(system, *symbols):
     # Return solutions
     solution = FiniteSet(tuple(solution))
     return solution
+
+
+##############################################################################
+# ------------------------------nonlinsolve ---------------------------------#
+##############################################################################
+
+
+def substitution(system, symbols, result, known_symbols, exclude, all_symbols):
+    r"""
+    Solves the `system` using substitution method.
+
+    Parameters
+    ==========
+
+    system : list of equations.
+
+    symbols : list of unsolved symbols.
+
+    known_symbols : list of solved symbols.
+
+    result : If non of the symbols are solved then it is empty list, otherwise
+    list of already solved_syms symbols( dict symbol : value )
+
+    exclude : Set of expression which should not be zero with result.
```

Comment:
Please don't assume anything while documenting your code for the developer who would read this. It should make proper sense and write everything explicitly, what do you mean by the statement.